src/kaleidoscope/progmem_helpers.h

```diff
@@ -0,0 +1,33 @@
+// -*- mode: c++ -*-
+/* Kaleidoscope - Firmware for computer input devices
+ * Copyright (C) 2013-2019  Keyboard.io, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+// Load any data type stored in PROGMEM into an object of that type in memory.
+template <typename _Type>
+void loadFromProgmem(_Type const& pgm_object, _Type& object) {
```

Comment:
Very useful helper functions!

You might want to add to the comments that these are only applicable to intrinsic data types and [trivial classes](https://stackoverflow.com/questions/4178175/what-are-aggregates-and-pods-and-how-why-are-they-special/7189821#7189821).

Reply:
I would be happy to write more detail about what types can and cannot be used…if I understood what the rules were. No inheritance? No virtual functions? No constructors that have a body other than `{}`?

This has worked properly for all of the classes I've used it with, and but I don't understand the seemingly arbitrary rules from the language specification.